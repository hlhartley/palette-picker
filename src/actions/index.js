export const setError = (error) => ({
  type: 'SET_ERROR', 
  error
});

export const setLoading = (isLoading) => ({
  type: 'SET_LOADING', 
  isLoading
});

export const setProjects = (projects) => ({
  type: 'SET_PROJECTS', 
  projects
});

export const setPalettes = (palettes) => ({
  type: 'SET_PALETTES', 
  palettes
});

export const setCurrentPaletteId = (paletteId) => ({
  type: 'SET_CURRENT_PALETTE_ID', 
  paletteId
});

export const setCurrentProjectName = (projectName) => ({
  type: 'SET_CURRENT_PROJECT_NAME', 
  projectName
});

export const setRandomHexcodes = (hexcodes) => ({
  type: 'SET_RANDOM_HEXCODES', 
  hexcodes
});

export const toggleDisplayRandom = (shouldDisplay) => ({
  type: 'DISPLAY_RANDOM', 
  shouldDisplay
});
