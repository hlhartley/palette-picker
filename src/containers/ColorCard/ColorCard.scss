.ColorCard {
    border: 1px solid purple;
    padding: 10px;
}

.lock-icon {
    color: grey;
    position: relative;
    left: 50%;
}