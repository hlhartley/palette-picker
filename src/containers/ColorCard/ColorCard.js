import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLock } from '@fortawesome/free-solid-svg-icons';
import { faUnlock } from '@fortawesome/free-solid-svg-icons';
import { connect } from 'react-redux';
import { setHexcodes } from '../../actions';

export class ColorCard extends Component {

  toggleColorLock = async () => {
    let updatedHexcodes = this.props.hexcodes.map((hexcode) => {
      if(hexcode.color === this.props.hexcode.color && hexcode.isLocked === true){
        hexcode.isLocked = false;
      } else if (hexcode.color === this.props.hexcode.color) {
        hexcode.isLocked = true;
      }
      return hexcode;
    })
    await this.props.setHexcodes(updatedHexcodes)
  }
  
  render() {
    return (
      <div className="ColorCard" style={{backgroundColor: `#${this.props.hexcode.color}`}}>
        <FontAwesomeIcon icon={this.props.hexcode.isLocked ? faLock : faUnlock} className="lock-icon" onClick={(e) => this.toggleColorLock(e)}/>
        <p className="hexcode">#{this.props.hexcode.color}</p>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  hexcodes: state.hexcodes
});

export const mapDispatchToProps = (dispatch) => ({
  setHexcodes: (hexcodes) => dispatch(setHexcodes(hexcodes))
});


export default connect(mapStateToProps, mapDispatchToProps)(ColorCard);

