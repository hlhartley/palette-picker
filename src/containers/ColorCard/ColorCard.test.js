import React from 'react';
import { shallow } from 'enzyme';
import { ColorCard, mapStateToProps, mapDispatchToProps } from './ColorCard';
import { setLockedHexcodes, removeLockedHexcode } from '../../actions';

describe('ColorCard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <ColorCard/>
      )
    });
    
  describe('ColorCard component', () => {
    it('should properly render the component elements', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with lockedHexcodes', () => {
      const mockState = {
        projects: [],
        isLoading: false,
        error: '',
        palettes: [],
        extra: '',
        lockedHexcodes: []
      }
      const expected = {
        lockedHexcodes: []
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });

    // **test toggleColorLock method here**
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when setLockedHexcodes is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = setLockedHexcodes();
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setLockedHexcodes()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('should call dispatch when removeLockedHexcode is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = removeLockedHexcode();
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.removeLockedHexcode()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
});