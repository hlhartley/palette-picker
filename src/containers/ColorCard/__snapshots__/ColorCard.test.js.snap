// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard ColorCard component should properly render the component elements 1`] = `
<div
  className="ColorCard"
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <FontAwesomeIcon
    border={false}
    className="lock-icon"
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f09c",
          "M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z",
        ],
        "iconName": "unlock",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    onClick={[Function]}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <p
    className="hexcode"
  >
    #
    ffffff
  </p>
</div>
`;
